hermes-common/src/main/java/pl/allegro/tech/hermes/common/message/wrapper/AvroMessageContentWrapper.java

```diff
@@ -23,20 +25,23 @@ UnwrappedMessageContent unwrapContent(byte[] data, Schema schema) {
             return new UnwrappedMessageContent(
                 new MessageMetadata(
                     Long.parseLong(metadata.get(METADATA_TIMESTAMP_KEY).toString()),
-                    metadata.get(METADATA_MESSAGE_ID_KEY).toString()),
+                    metadata.get(METADATA_MESSAGE_ID_KEY).toString(),
+                    ofNullable(metadata.get(METADATA_TRACE_ID_KEY)).map(Object::toString).orElse(null)),
                 data);
         } catch (Exception exception) {
             throw new UnwrappingException("Could not read avro message", exception);
         }
     }
 
-    byte[] wrapContent(byte[] message, String id, long timestamp, Schema schema) {
+    byte[] wrapContent(byte[] message, String id, String traceId, long timestamp, Schema schema) {
         try {
             GenericRecord genericRecord = bytesToRecord(message, schema);
-            genericRecord.put(METADATA_MARKER, of(
-                METADATA_TIMESTAMP_KEY, Long.toString(timestamp),
-                METADATA_MESSAGE_ID_KEY, id
-            ));
+            Map<Utf8, String> metadata = Maps.newHashMap();
+            metadata.put(METADATA_TIMESTAMP_KEY, Long.toString(timestamp));
+            metadata.put(METADATA_MESSAGE_ID_KEY, id);
+            ofNullable(traceId).ifPresent((val) -> metadata.put(METADATA_TRACE_ID_KEY, val));
```

Comment:
Yes it might be absent, and at this point we do not generate any value instead, which make sense since the trace id is a notion of some external id.